### PR TITLE
Feat: Navigation Item component

### PR DESCRIPTION
**Pull request by abu-seini-maf**
added new Nav Item Component.
modified NAV Component.
Updated autoloader.

### CODE REVIEW

File: src/components/mf-nav-item/mf-nav-item.styles.ts [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    position: relative;
+    display: flex;
+    align-items: stretch;
+    font-family: var(--sl-font-sans);
+    font-size: var(--sl-font-size-medium);
+    font-weight: var(--sl-font-weight-normal);
+    line-height: var(--sl-line-height-normal);
+    letter-spacing: var(--sl-letter-spacing-normal);
+    color: var(--sl-color-neutral-700);
+    padding: var(--sl-spacing-2x-small) var(--sl-spacing-2x-small);
```
Comment (carlos-verdes) [inline src/components/mf-nav-item/mf-nav-item.styles.ts:24]: let's use this for naming convention:

```css
.nav-item {
   /* ... */
  padding: var(--nav-spacing-inset-stretch, --nav-spacing-inset) var(--nav-spacing-inset-squish, --nav-spacing-inset);
  margin-right: var(--nav-spacing-inline);
  margin-bottom: var(--nav-spacing-stack);
}

.nav-item:dir(rtl) {
  margin-left: var(--nav-spacing-inline);
  margin-right: 0px !important;
}



<img width="756" alt="image" src="https://github.com/user-attachments/assets/e6d1aff4-550d-4635-9311-6c4580818692">

File: src/components/mf-nav-item/mf-nav-item.styles.ts [hunk 2]
```diff
@@ -0,0 +1,161 @@
+import { css } from 'lit';
```
Comment (carlos-verdes) [inline src/components/mf-nav-item/mf-nav-item.styles.ts:1]: I understand we need to add lot of custom variables at component level, I just added example for nav padding and margin following the space naming conventions I shared before

File: src/components/mf-nav-item/mf-nav-item.test.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import '../../../dist/shoelace.js';
+import { expect, fixture, html } from '@open-wc/testing';
+
+describe('<mf-nav-item>', () => {
+  it('should render a component', async () => {
+    const el = await fixture(html` <mf-nav-item></mf-nav-item> `);
+
+    expect(el).to.exist;
+  });
+});
```
Comment (carlos-verdes) [inline src/components/mf-nav-item/mf-nav-item.test.ts:10]: amazing test :)

**Review by carlos-verdes — COMMENTED** [review]
I left couple of comments, most of it don't need change (are just observations)
I didn't go deep into controller logic as I prefer to merge this PR fast and do incremental changes, first thing as we agreed to add more CSS variables.

Apply the changes you may think are needed and fell free to merge.

File: docs/pages/components/mf-nav-item.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+</mf-navigation>
+```
+
+{% raw %}
+
+## Examples
+
+### Prefix & Suffix
+
+Add content to the start and end of nav items using the `prefix` and `suffix` slots.
```
[the remark this answers is not in the document]
Comment (abu-seini-maf) [reply docs/pages/components/mf-nav-item.md:31]: following the same structure and behavior of Shoelace, adding it within CSS will limit the usage of the Icons lib.

File: docs/pages/components/mf-nav-item.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+```html:preview
+<mf-navigation style="min-height: 200px;">
+  <mf-nav-item>Option 1</mf-nav-item>
+  <mf-nav-item loading>Option 2</mf-nav-item>
+  <mf-nav-item>Option 3</mf-nav-item>
+</mf-navigation>
+```
+
+### Highlight
```
[the remark this answers is not in the document]
Comment (abu-seini-maf) [reply docs/pages/components/mf-nav-item.md:79]: as discussed highlight have a different logic than active.

File: docs/pages/components/mf-navigation.md [hunk 6]
```diff
@@ -5,19 +5,279 @@ meta:
 layout: component
 ---
 
+You can use [Nav items](/components/mf-nav-item) to compose a navigation. Navigation support keyboard interactions, including type-to-select an option.
```
[the remark this answers is not in the document]
Comment (abu-seini-maf) [reply docs/pages/components/mf-navigation.md:8]: after looking into the actual behavior of shoelace, we wouldn't want to use css grid for ordering.

File: docs/pages/components/mf-navigation.md [hunk 7]
```diff
@@ -5,19 +5,279 @@ meta:
...
+        <MfNavItem value="capitalize">Capitalize</MfNavItem>
+      </MfNavigation>
+    </MfNavItem>
+  </MfNavigation>
+);
+```
+
+{% endraw %}
+
+### Vertical Sub-Navigation
```
[the remark this answers is not in the document]
Comment (abu-seini-maf) [reply docs/pages/components/mf-navigation.md:217]: from a semantics / and best practices behavior, any navigation element should be used within an anchor / NAV and not using menu elements.

File: src/components/mf-nav-item/mf-nav-item.component.ts [hunk 8]
```diff
@@ -0,0 +1,174 @@
...
+  @property() type = 'normal';
+
+  /** Draws the nav item in a loading state. */
+  @property({ type: Boolean, reflect: true }) loading = false;
+
+  /** Draws the nav item in a disabled state, preventing selection. */
+  @property({ type: Boolean, reflect: true }) disabled = false;
+
+  /** Draws the nav item in a highlight state, provides custom class highlight. */
+  @property({ type: Boolean, reflect: true }) highlight = false;
```
[the remark this answers is not in the document]
Comment (abu-seini-maf) [reply src/components/mf-nav-item/mf-nav-item.component.ts:72]: as mentioned before, highlight have a different behavior from active

File: src/components/mf-nav-item/mf-nav-item.component.ts [hunk 9]
```diff
@@ -0,0 +1,174 @@
...
+
+    // When the label changes, emit a slotchange event so parent controls see it
+    if (textLabel !== this.cachedTextLabel) {
+      this.cachedTextLabel = textLabel;
+      this.emit('slotchange', { bubbles: true, composed: false, cancelable: false });
+    }
+  }
+
+  private handleHostClick = (event: MouseEvent) => {
+    // Prevent the click event from being emitted when the button is disabled or loading
```
[the remark this answers is not in the document]
Comment (abu-seini-maf) [reply src/components/mf-nav-item/mf-nav-item.component.ts:106]: couldn't understand.

File: src/components/mf-nav-item/mf-nav-item.component.ts [hunk 10]
```diff
@@ -0,0 +1,174 @@
...
+    const isRtl = this.localize.dir() === 'rtl';
+    const isSubnavExpanded = this.subnavController.isExpanded();
+
+    return html`
+      <a
+        id="anchor"
+        part="base"
+        class=${classMap({
+          'nav-item': true,
+          'nav-item--rtl': isRtl,
```
[the remark this answers is not in the document]
Comment (abu-seini-maf) [reply src/components/mf-nav-item/mf-nav-item.component.ts:148]: adding RTL class here as following shoelace behavior, and only adding classes for identification.

File: src/components/mf-nav-item/mf-nav-item.component.ts [hunk 11]
```diff
@@ -0,0 +1,174 @@
...
+          'nav-item--disabled': this.disabled,
+          'nav-item--loading': this.loading,
+          'nav-item--highlight': this.highlight,
+          'nav-item--has-subnav': this.isSubnav(),
+          'nav-item--subnav-expanded': isSubnavExpanded
+        })}
+        ?aria-haspopup="${this.isSubnav()}"
```
[the remark this answers is not in the document]
Comment (abu-seini-maf) [reply src/components/mf-nav-item/mf-nav-item.component.ts:155]: good thing to keep in mind.

File: src/components/mf-nav-item/mf-nav-item.component.ts [hunk 12]
```diff
@@ -0,0 +1,174 @@
...
+        ?aria-expanded="${isSubnavExpanded ? true : false}"
+        href="${this.href}"
+      >
+        <slot name="prefix" part="prefix" class="nav-item__prefix"></slot>
```
[the remark this answers is not in the document]
Comment (abu-seini-maf) [reply src/components/mf-nav-item/mf-nav-item.component.ts:159]: following shoelace structure and behavior.

File: src/components/mf-nav-item/mf-nav-item.component.ts [hunk 13]
```diff
@@ -0,0 +1,174 @@
...
+
+        <slot part="label" class="nav-item__label" @slotchange=${this.handleDefaultSlotChange}></slot>
+
+        <slot name="suffix" part="suffix" class="nav-item__suffix"></slot>
```
[the remark this answers is not in the document]
Comment (abu-seini-maf) [reply src/components/mf-nav-item/mf-nav-item.component.ts:163]: following shoelace structure and behavior.

File: src/components/mf-nav-item/mf-nav-item.styles.ts [hunk 14]
```diff
@@ -0,0 +1,161 @@
...
+
+export default css`
+  :host {
+    --subnav-offset: -2px;
```
[the remark this answers is not in the document]
Comment (abu-seini-maf) [reply src/components/mf-nav-item/mf-nav-item.styles.ts:5]: this to align the subnav with the parent while mitigating the space from borders.

**Pull request closed without merging**